### PR TITLE
Transforms update meta

### PR DESCRIPTION
**Pull request by rijobro**
Notebook demonstrating how transforms update meta data and results can be viewed in external viewer without the need for resampling or inverse transformations. Input and output images have different sizes and spacing.

image

@wyli is this what you had in mind?

### Status
**Ready**

### Checks

- [x] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

**Comment by wyli** [comment]
thanks, this is very nice

- with the metatensor we are able to apply Spacing anywhere in the preprocessing.

- cc @Nic-Ma this one line in the tutorial is very straightforward:

```py
[post_trans(i) for i in decollate_batch(model(batch["image"].to(device))) for batch in dl]
```

(I'm still working on more unit tests with the metatensor feature branch)

**Comment by wyli** [comment]
perhaps to visualise it natively in the notebook later we might want to look into https://github.com/InsightSoftwareConsortium/itkwidgets

**Comment by rijobro** [comment]
> perhaps to visualise it natively in the notebook later we might want to look into https://github.com/InsightSoftwareConsortium/itkwidgets

I looked into this, but unfortunately it requires the image and label to be the same size as each other.

File: modules/transforms_update_meta_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,297 @@
+{
```
Comment (Nic-Ma) [inline modules/transforms_update_meta_data.ipynb:1]: <ol><li>"child class" -&gt; "sub class", or "inherit from <code>torch.Tensor</code>".</li><li>Could you please help add the "open in colab" button? Refer to: https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb</li></ol><p> </p><p>Thanks.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/758/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (Nic-Ma) [inline modules/transforms_update_meta_data.ipynb:1]: <p>Could you please help also add the install logic like other tutorial notebooks?</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/758/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Nic-Ma) [inline modules/transforms_update_meta_data.ipynb:1]: <p>Better to briefly introduce the dataset used here.</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/758/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Review by Nic-Ma — APPROVED** [review]

**Comment by wyli** [comment]
> > perhaps to visualise it natively in the notebook later we might want to look into https://github.com/InsightSoftwareConsortium/itkwidgets
> 
> I looked into this, but unfortunately it requires the image and label to be the same size as each other.

cc @thewtex @aylward display images in physical space at different resolutions/orientations/sizes could be a feature request to itkwidgets?

**Comment by rijobro** [comment]
@Nic-Ma thanks for the review, I've incorporated your suggestions.

**Comment by aylward** [comment]
Great idea...we can definitely add this feature, but in this demo, wouldn't it be better to show how to get the label back into the original resolution (e.g., via inverted transforms or updating the resize transform to accept a target image that defines the desired size and spacing)?   Unless the segmentation is purely for visualization or the ideal segmentation was defined at the lower resolution?

**Comment by aylward** [comment]
@wyli - https://github.com/InsightSoftwareConsortium/itkwidgets/issues/451

**Comment by wyli** [comment]
thanks @aylward, this demo is mainly to show that all the ROIs sampled from the input volume have their metadata (in this case the voxel to physical space transform) properly handled throughout the deep learning pipeline. so, after the model inference, the output of the ROIs could be displayed in physical space together with the input volume. mainly as a visual example showcasing the MetaTensor...

**Comment by rijobro** [comment]
Yes @aylward we have other tutorials where we demonstrate the inverse (e.g., [here](https://github.com/Project-MONAI/tutorials/blob/main/modules/inverse_transforms_and_test_time_augmentations.ipynb)). The objective of this one is to show that things line up regardless of sizes and spacing, which is obviously accentuated when there is a mismatch.

**Comment by aylward** [comment]
Ah - makes sense!   Thanks!

**Comment by thewtex** [comment]
CC @PaulHax

**Comment by moonforsun** [comment]
This is nice for the visualization purpose to align the segmentation the original image despite different sizes. I am wondering any following-up update for the meta data after transforms? I am using MONAI 1.1.0 and the meta data is not updated after e.g. a `monai.transforms.SpatialCrop`.

**Comment by wyli** [comment]
Hi @moonforsun, yes please see the demo about this topic https://github.com/Project-MONAI/tutorials/blob/main/modules/transforms_metatensor.ipynb

**Pull request merged**